src/main/java/com/epam/training/circularbuffer/CircularBuffer.java

```diff
@@ -0,0 +1,94 @@
+package com.epam.training.circularbuffer;
+
+import com.epam.training.exception.NotEnoughSpaceException;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+public class CircularBuffer<T> {
+    private Object[] data;
+    private int initialCapacity;
+    private int headPosition = 0;
+    private int tailPosition = 0;
+    private int size = 0;
+
+    public CircularBuffer(int initialCapacity) {
+        this.initialCapacity = initialCapacity;
+        this.data = new Object[initialCapacity];
+    }
+
+    public void put(T t) {
+        if (isFull()) {
+            throw new RuntimeException("The buffer is full!");
+        }
+        data[headPosition] = t;
+        size++;
+        headPosition = nextIndexInACircle(headPosition);
+    }
+
+    public T get() {
+        if (tailPosition == headPosition && !isFull()) {
+            throw new RuntimeException("The buffer is empty!");
```

Comment:
Do not throw RuntimeException. Use the child of Runtime or you created.

Reply:
done